resources/assets/js/components/JobTasks/JobTasks.tsx

```diff
@@ -1,222 +1,387 @@
+                data-c-border="all(thin, solid, grey)"
+                data-c-alignment="centre"
+              >
+                <FormattedMessage
+                  id="jobTasks.taskCount.none"
+                  defaultMessage="You don't have any tasks added yet!"
+                  description="Message displayed when there are no tasks present on the page."
+                />
               </div>
-            </div>
-          </div>
-          <div
-            className="job-builder-task invalid"
-            data-c-input="textarea"
-            data-c-grid-item="base(1of1)"
-          >
-            <div>
-              <button type="button" data-tc-builder-task-up-trigger>
-                <i className="fas fa-long-arrow-alt-up" />
-              </button>
-              <button type="button" data-tc-builder-task-down-trigger>
-                <i className="fas fa-long-arrow-alt-down" />
-              </button>
-            </div>
-            <div>
-              <button type="button" data-tc-builder-task-delete-trigger>
-                <i className="fas fa-trash" data-c-colour="stop" />
-              </button>
-            </div>
-            <label htmlFor="TA2">Task 03</label>
-            <span>Required</span>
-            <div>
-              <textarea
-                id="TA2"
-                placeholder="Try for a casual, frank, friendly tone..."
-              />
-            </div>
-            <span>This input has an error.</span>
-          </div>
-          <div
-            className="job-builder-task invalid"
-            data-c-input="textarea"
-            data-c-grid-item="base(1of1)"
-          >
-            <div>
-              <button type="button" data-tc-builder-task-up-trigger>
-                <i className="fas fa-long-arrow-alt-up" />
-              </button>
-              <button type="button" data-tc-builder-task-down-trigger>
-                <i className="fas fa-long-arrow-alt-down" />
-              </button>
-            </div>
-            <div>
-              <button type="button" data-tc-builder-task-delete-trigger>
-                <i className="fas fa-trash" data-c-colour="stop" />
-              </button>
-            </div>
-            <label htmlFor="TA2">Task 04</label>
-            <span>Required</span>
-            <div>
-              <textarea
-                id="TA2"
-                placeholder="Try for a casual, frank, friendly tone..."
+            )}
+            <Form id="job-tasks">
+              <FieldArray
+                name="tasks"
+                render={(arrayHelpers): React.ReactElement => (
+                  <>
+                    <div data-c-grid="gutter">
+                      {values.tasks &&
+                        values.tasks.length > 0 &&
+                        values.tasks.map(
+                          (task, index): React.ReactElement => (
+                            <>
+                              {validCount! === index && (
```

Comment:
Why the non-null assertions on validCount, here and later? Don't we need to account for it possibly being undefined?

Reply:
After reviewing the functionality for the page, I'm making it required. I'll take out the non-nulls.